### PR TITLE
feat(backend, mock-ase): rename payment pointer to wallet address in rafiki code

### LINKED ISSUE #1935: Backend: rename 'payment pointer' to 'wallet address' in backend code
Remove all references to 'payment pointer' in the Rafiki backend code and replace it with 'wallet address'.

### PR DESCRIPTION
**Pull request by njlie**
## Changes proposed in this pull request

- Renames all instances of `paymentPointer` and the like to `walletAddress` and the like in the Rafiki codebase. 

## Context

Fixes #1935.

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass
- [ ] Postman collection updated

### CODE REVIEW

**Comment by njlie** [comment]
I would recommend anyone reviewing it just try to make sure no instances of "payment pointer" or any such variant appear in the rafiki code. It'll be more valuable to rigorously test this once it is in `nl-open-payments` and it builds.

**Review by mkurapov — COMMENTED** [review]
Just the same comment as the #1951, just need to run `generate:types` in token-introspection

File: packages/backend/migrations/20230918113102_rename_payment_pointer_tables.js [hunk 1]
```diff
@@ -40,21 +40,27 @@ exports.up = function (knex) {
       table.dropIndex(['paymentPointerId', 'createdAt', 'id'])
       table.renameColumn('paymentPointerId', 'walletAddressId')
       table.foreign('walletAddressId').references('walletAddresses.id')
-      table.index(['walletAddressId', 'createdAt', 'id'])
+      table.index(['walletAddressId'])
+      table.index(['createdAt'])
+      table.index(['id'])
```
Comment (BlairCurrey) [inline packages/backend/migrations/20230918113102_rename_payment_pointer_tables.js:45]: I think these (and similar) should be re-added as a composite index instead? https://github.com/interledger/rafiki/pull/1937#discussion_r1357183892

File: localenv/mock-account-servicing-entity/app/lib/accounts.server.ts [hunk 2]
```diff
@@ -15,10 +15,10 @@ export interface Account {
 
 export interface AccountsServer {
   clearAccounts(): Promise<void>
-  setPaymentPointer(
+  setWalletAddress(
     id: string,
     pointerID: string,
-    paymentPointer: string
+    walletAddress: string
   ): Promise<void>
```
Comment (BlairCurrey) [inline localenv/mock-account-servicing-entity/app/lib/accounts.server.ts:22]: looks like `pointerID` is short for `paymentPointerID` and should be renamed as well

File: packages/backend/src/payment-method/ilp/service.test.ts [hunk 3]
```diff
@@ -7,11 +7,11 @@ import { IAppConfig, Config } from '../../config/app'
 import { IocContract } from '@adonisjs/fold'
 import { AppServices } from '../../app'
 import { createAsset } from '../../tests/asset'
-import { createPaymentPointer } from '../../tests/paymentPointer'
+import { createWalletAddress } from '../../tests/walletAddress'
 import { Asset } from '../../asset/model'
 import { withConfigOverride } from '../../tests/helpers'
 import { StartQuoteOptions } from '../handler/service'
-import { PaymentPointer } from '../../open_payments/payment_pointer/model'
+import { WalletAddress } from '../../open_payments/payment_pointer/model'
```
Comment (BlairCurrey) [inline packages/backend/src/payment-method/ilp/service.test.ts:14]:
```suggestion
import { WalletAddress } from '../../open_payments/wallet_address/model'
```

**Review by BlairCurrey — COMMENTED** [review]
another place to update: 
https://github.com/interledger/rafiki/blob/9338895c036285c60f282afbd34ad88f999e1c5f/packages/frontend/app/shared/enums.ts#L8

File: packages/backend/migrations/20230918113102_rename_payment_pointer_tables.js (hunk 1 above)
Comment (njlie) [reply packages/backend/migrations/20230918113102_rename_payment_pointer_tables.js:45]: This hasn't been rebased off of the feature branch yet, so it doesn't have the latest changes I made to the db table migration. Should go away once I do that.

**Comment by mkurapov** [comment]
@njlie just some formatting

File: localenv/mock-account-servicing-entity/app/lib/accounts.server.ts (hunk 2 above)
Comment (BlairCurrey) [reply localenv/mock-account-servicing-entity/app/lib/accounts.server.ts:22]: @njlie I think this one still needs renaming

**Review by mkurapov — APPROVED** [review]
If the test fixes will happen separate and the `20230918113102_rename_payment_pointer_tables` migration is fixed in #2043 then its good to go

**Review by BlairCurrey — APPROVED** [review]
looks good to me... cant find any more variations of payment pointer.

**Pull request merged**